ios/ElectronCash/electroncash_gui/ios_native/send.py

```diff
@@ -402,7 +412,7 @@ def textFieldShouldReturn_(self, tf : ObjCInstance) -> bool:
         return True
 
     @objc_method
-    def onPayTo_message_amount_(self, address, message, amount) -> None:
+    def onPayTo_message_amount_(self, address, message, amount, op_return, op_return_raw) -> None:
```

Comment:
Hmm. Did you ever wonder why these methods have such funny names? They are designed to follow the objective-c naming convention which tries to do a sort of poor-man’s named args by mangling the name. So this method as it was originally written (before your change) would get called like this in objc:


```objc
[someObj onPayTo:arg1 message:arg2 amount:arg3];
```

the typical way is to provide “overload” for additional versions that add more args.. so the new method should read something like:

```objc
[someObj onPayTo:arg1 message:arg2 amount:arg3 opreturn:arg4 isRaw:YES]:
```

and the 3 arg version calls the 4 or 5 arg version providing it with default arguments.

mind to objective-c has been around since the early 90s and back then this was considered a very ergonomic language.

Anyway what’s my point here? Well maybe we should stick to that convention. Right now the method can’t easily be called from objc I think, oddly.

it may not matter in this app — this is largely a python app anyway — but maybe we can stick to the convention?

I’m on my phone now though so it’s hard to write more. I can suggest changes later when I get to my computer...

Reply:
In python then, the method signature would be:

```python
@objc_method
def onPayTo_message_amount_opreturn_isRaw_(self, address, message, amount, opreturn, israw):
```

note that I’m suggesting we also collapse the opreturn arg into 1, and provide a Boolean for how to interpret it.

then you keep the old python method that takes 3 params and just call the new one with `, None, False)` as the 4th and 5th args...

this way the objc convention is preserved ....

Reply:
Also in my way ... in case you missed a spot where you forgot to update the method call .. it still “works”.

I know it’s awkward but such is the objc way...

Reply:
I understand. Thanks for the insight, I will change it.